### PR TITLE
sched/task: Implement gettid(2) syscall

### PR DESCRIPTION
**Pull request by anchao**
## Summary

sched/task: Implement gettid(2) syscall

See the reference here:
https://man7.org/linux/man-pages/man2/gettid.2.html

Change-Id: Ia814d0ccc3b20d8dfc36c809682ddf6e21811d20
Signed-off-by: chao.an <anchao@xiaomi.com>

## Impact

add get thread id support, which is no different as process id on NuttX os

## Testing

API Testing

### CODE REVIEW

**Review by xiaoxiang781216 — APPROVED** [review]
LGTM.

**Comment by yamt** [comment]
do you plan to "fix" getpid? otherwise this doesn't make sense.

File: sched/task/task_gettid.c [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+ *
+ * Returned Value:
+ *   On success, returns the thread ID of the calling process.
+ *
+ ****************************************************************************/
+
+pid_t gettid(void)
+{
+  return getpid();
+}
```
Comment (patacongo) [inline sched/task/task_gettid.c:49]: This really belongs in libs/libc/unistd.  It is wasteful to have two system calls that do the same thing.

Comment (patacongo) [reply sched/task/task_gettid.c:49]: See PR #1873

**Pull request merged**